packages/@react-spectrum/utils/src/Slots.tsx

```diff
@@ -28,6 +28,26 @@ export function cssModuleToSlots(cssModule) {
   }, {});
 }
 
+export function SlotProvider(props) {
+  let parentSlots = useContext(SlotContext) || {};
```

Comment:
are we worried about merging slots from parents down? would it be better to be choosey and have components only merge specific things they need? I'm guessing this came up due to button now have its own slots context that was wiping out other ones?
maybe it's enough that we have clear slots though, so people can still be choosey if they want to

Reply:
so, I needed to combine class names from Picker and FieldButton for the icon. I could have done it manually, but figured it might be nice to be able to compose slot props together like this. We might run into a case where this isn't desired, but we could use `<ClearSlots>` to fix that.